internal/templates/data/common/js/cursor/sdk.mdc

```diff
@@ -0,0 +1,92 @@
+---
+description: Agentuity JavaScript SDK API Reference
+globs: "**/*.{js,ts,jsx,tsx}"
+---
+
+# Agentuity JavaScript SDK
+
+The Agentuity JavaScript SDK provides a powerful framework for building AI agents in JavaScript and TypeScript. This cursor rules file helps you navigate the SDK's core interfaces and methods.
+
+## Core Interfaces
+
+### AgentHandler
+
+The main handler function type for an agent:
+
+```typescript
+type AgentHandler = (
+  request: AgentRequest,
+  response: AgentResponse,
+  context: AgentContext
+) => Promise<AgentResponseType>;
+```
+
+### AgentRequest
+
+The `AgentRequest` interface provides methods for accessing request data:
+
+- `request.trigger`: Gets the trigger type of the request
+- `request.metadata(key, defaultValue)`: Gets metadata associated with the request
+- `request.data.json`: Gets the payload as a JSON object
+- `request.data.text`: Gets the payload as a string
+- `request.binary()`: Gets the payload as an ArrayBuffer
+
+### AgentResponse
+
+The `AgentResponse` interface provides methods for creating responses:
+
+- `response.json(data, metadata)`: Creates a JSON response
+- `response.text(data, metadata)`: Creates a text response
+- `response.binary(data, metadata)`: Creates a binary response
+- `response.html(data, metadata)`: Creates an HTML response
+- `response.empty(metadata)`: Creates an empty response
+- `response.handoff(agent, payload, contentType, metadata)`: Redirects to another agent
+
+### AgentContext
+
+The `AgentContext` interface provides access to various capabilities:
+
+- `context.logger`: Logging functionality
+- `context.kv`: Key-Value storage
+- `context.vector`: Vector storage
+- `context.getAgent(params)`: Gets a handle to a remote agent
+- `context.tracer`: OpenTelemetry tracing
+
+## Storage APIs
+
+### Key-Value Storage
+
+Access through `context.kv`:
+
+- `context.kv.get(name, key)`: Retrieves a value
+- `context.kv.set(name, key, value, ttl)`: Stores a value
```

Comment:
instead of ttl it should be params using type `KeyValueStorageSetParams`

```typescript
export interface KeyValueStorageSetParams {
	ttl?: number;
	contentType?: string;
}
```